src/main/java/Duke.java

```diff
@@ -1,10 +1,234 @@
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    public static final String FILE_PATH = "duke.txt";
+    static String lineCutOff = "_______________________";
+    static int TODO_TASK_INDEX = 5;
+    static int DEADLINE_TASK_INDEX = 9;
+    static int EVENT_TASK_INDEX = 6;
+    static int DONE_TASK_INDEX = 5;
+    static int DELETE_TASK_INDEX = 7;
+
+    //save changes
+    public static void writeToFile(String FILE_PATH){
+        try{
+            File f = new File(FILE_PATH);
+            FileWriter fw = new FileWriter(FILE_PATH);
+            for (Task task: tasks){
+                if(task != null)
+                fw.write(task.writeToFile());
+            }
+            fw.close();
+        } catch (IOException e){
+            System.out.println("Something went wrong with IO stream.\n");
+        }
+    }
+    
+    public static void readFromFile(String FILE_PATH){
+        try{
+            File f = new File(FILE_PATH);
+            Scanner s = new Scanner(f);
+            Task task;
+            while(s.hasNext()){
+                String[] descriptions = s.nextLine().split("\\|");
+                if (descriptions[0].equals("T")) {
+                    task = new ToDo(descriptions[2]);
+                } else if (descriptions[0].equals("D")) {
+                    task = new Deadline(descriptions[2], descriptions[3]);
+                } else if (descriptions[0].equals("E")) {
+                    task = new Event(descriptions[2], descriptions[3]);
+                } else {
+                    tasks.add(null);
+                    break;
+                }
+
+                if (descriptions[1].equals("1")){
+                    task.setIsDone(true);
+                }
+                tasks.add(task);
+            }
+        } catch (FileNotFoundException e){
+            System.out.println("File not found");
+        }
+    }
+
+    public static Task taskType(String task) {
+        try {
+            if (task.startsWith("todo")) {
+                if (task.equals("todo")) {
+                    throw new DukeException();
+                }
+                return new ToDo(task.substring(TODO_TASK_INDEX));
+            } else if (task.startsWith("deadline")) {
+                int indexBy = task.indexOf("/");
+                if (task.equals("deadline")) {
+                    throw new DukeException();
+                } else if (indexBy== -1) {
+                    throw new DukeException();
+                }
+                return new Deadline(task.substring(DEADLINE_TASK_INDEX, indexBy - 1), task.substring(indexBy + 3));
```

Comment:
Do note using of magic literals, perhaps you could initialize some constants int with descriptive name to help users understand these numbers better.